src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs

```diff
@@ -490,13 +490,19 @@ private IEnumerable<Location> GetDiagnosticLocationsWithinTree(SyntaxTree tree)
             }
         }
 
-        internal Diagnostic WithReportDiagnostic(ReportDiagnostic reportAction)
+        internal Diagnostic WithReportDiagnostic(ReportDiagnostic reportAction, bool reportSuppressed)
```

Comment:
Why "reportSuppressed" instead of "isSuppresed"? Latter seemed more intuitive as it's a direct mapping to a property being set.

Reply:
The diagnostic may be suppressed regardless. Specifically, the value of ReportDiagnostic could be suppressed regardless of what the boolean is. The difference is in the setting of whether or not we report suppressed diagnostics. Technically, the full name should be `reportSuppressedDiagnostics` but that seemed verbose. Would you prefer that? I think `isSuppressed` is a bit confusing because then it seems like it contradicts `reportAction` in some cases.